forge/db/models/StorageSession.js

```diff
@@ -20,6 +20,15 @@ module.exports = {
                         where: { ProjectId: project },
                         attributes: ['id', 'sessions']
                     })
+                },
+                byUsername: async (username) => {
+                    const allStorageSessions = await this.findAll({ attributes: ['sessions', 'ProjectId'] })
```

Comment:
This won't scale very well when there are a large number of projects.

There are two optimisations of differing complexity:

1. get a list of projects the user has access to, then only get the storage sessions for those projects.
    That is actually quite a lot of DB work

2. optimise this findAll with a `where` query that `sessions` contains the text `username`. That will reduce the amount of data that has to be retrieved from the database in the one query.

Reply:
I took the path of least resistance as I suspected it would lead to discussion - possibly ending up on a change to DB schema that would scale much better?  However, I bet that is something we want to avoid unless there is no other solution?  I will implement No2 (unless you ping me otherwise? :) )